kyuubi-server/web-ui/package.json

```diff
@@ -0,0 +1,50 @@
+{
+  "name": "kyuubi-ui",
+  "private": true,
+  "version": "1.6.0-SNAPSHOT",
+  "type": "module",
+  "scripts": {
+    "dev": "vite --port 8080",
+    "build": "vue-tsc --noEmit && vite build --mode production",
+    "build-check": "npm run lint && npm run build",
+    "preview": "vite preview",
+    "lint": "eslint --ext .ts,vue --ignore-path .gitignore .",
+    "lint-fix": "eslint --fix --ext .ts,vue --ignore-path .gitignore .",
+    "prettier": "prettier --write \"src/**/*.{vue,ts,tsx}\"",
+    "test": "vitest",
+    "coverage": "vitest run --coverage"
+  },
+  "dependencies": {
+    "@element-plus/icons-vue": "^2.0.9",
+    "axios": "^0.27.2",
+    "element-plus": "^2.2.12",
+    "pinia": "^2.0.18",
+    "pinia-plugin-persistedstate": "^2.1.1",
+    "vue": "^3.2.37",
+    "vue-i18n": "^9.2.2",
+    "vue-router": "^4.1.3"
```

Comment:
what if we use the exactly version instead of a "^" at the begining?

Reply:
My understanding is that using a fixed version is very unprofessional behavior, package.json version has it's rules, such as ’*‘, '~', '^',    '^' is a default behavior by we enter npm install vue-router.  '^' has it's rules, for exampe "vue": "^3.2.37",  if vue team update a new version to 3.2.40,  the last number always means fix serveral bugs, so next time we build our project, our project will install 3.2.40. Also if vue team update to 3.3.0, the middle number means add some features / functions, our project also will install 3.3.0, but if vue team publush to 4.0.0, ^ will limit our project to install newest 3.X.X. But to keep save, I refer to the dol project, I also pull the package-lock.json to github, this file will keep only install the fixed version, package-lock.json is more installation and convenient than hard coding version....
Here is my learning of package.json, just a reference:
https://github.com/tongwl/front-end-knowledge/blob/main/node/%E5%85%B3%E4%BA%8Enpm%E7%9A%84package.json%EF%BC%8C%E4%BD%A0%E7%9F%A5%E9%81%93%E5%A4%9A%E5%B0%91%3F.md

Reply:
I've heard of those things. It recommends using dynamic versions to benefit patches applied automatically, but this causes the dependency tree to become indeterminate, then it recommends using `package.lock.json` to pin versions to restore determinism.
Sounds like it create a problem first, then create a new tool to solve the problem...

Reply:
Yes, you are right. 👍  While ensuring that the installation will not go wrong, we can update package-lock.json from time to time to ensure the update of our dependent plug-ins, as plug-ins also have bugs and new features, and the bugs of plug-ins can be repaired themselves, we can also enjoy the new features from the plugin.

Reply:
The vision is ideal, but the fact is rude, the determinism is much much important for a project. Anyway, let's follow the community common practices to declare the dynamic version and pin dependency tree in `package-lock.json`.